dependency list

The documentation does not contain a dependency list. It would be useful.

I had to install the glew headers (`libglew-dev` on debian-based systems), but I don't know what other requirements there are, other than a C compiler.

> @pnorman is this still an issue? npm install should cover all deps from the looks of it.

I tested and `npm install && npm start` now works on Ubuntu 14.04 with only git, npm and nodejs-legacy installed.

> However, to run the headless rendering tests, libglew-dev and imagemagick are required.

On the same VM I tried `npm test` with git, npm, nodejs-legacy, libglew-dev, imagemagick, and pastebinit (to get the text out of the vm) installed.

I got the error
```

> mapbox-gl@0.7.0 test /home/pnorman/mapbox-gl-js
> npm run lint && tape test/js/*/*.js


> mapbox-gl@0.7.0 lint /home/pnorman/mapbox-gl-js
> eslint js test


module.js:340
    throw err;
          ^
Error: Cannot find module 'gl'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/pnorman/mapbox-gl-js/js/util/canvas.js:6:10)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
npm ERR! weird error 8
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
npm ERR! not ok code 0
```

--- a/README.md
+++ b/README.md
@@ -17,7 +17,20 @@ Note that `mapbox-gl-js` is [currently unsafe](https://github.com/mapbox/mapbox-
 
 ## Developing mapbox-gl-js
 
-On linux, libglew-dev is required in order to run rendering tests:
+The following tools are required on any platform to develop `mapbox-gl-js`.
+Mac users are advised to use Homebrew unless they want to build these packages
+manually. APT install steps are relevant to Ubuntu Linux users.
+
+* [git](https://git-scm.com/)
+  * OSX: `brew install git`
+  * APT: `apt-get install git`
+* [node.js](https://nodejs.org/)
+* [GNU Make](http://www.gnu.org/software/make/)
+* [imagemagick](http://www.imagemagick.org/)
+  * OSX: `brew install imagemagick`
+  * APT: `apt-get install imagemagik`
+
+On Linux, libglew-dev is required in order to run rendering tests:
 
 ```
 $ sudo apt-get install libglew-dev

